relation-extraction/src/main/java/org/cogcomp/re/RelationAnnotator.java

```diff
@@ -61,61 +72,64 @@ public void initialize(ResourceManager rm) {
             WordNetManager.loadConfigAsClasspathResource(true);
             wordNet = WordNetManager.getInstance();
             gazetteers = GazetteersFactory.get();
-        }
-        catch (Exception e){
+        } catch (Exception e) {
             e.printStackTrace();
         }
     }
 
     @Override
     public void addView(TextAnnotation record) throws AnnotatorException {
-        if (!isInitialized()){
+        if (!isInitialized()) {
             doInitialize();
         }
-        if (!record.hasView(ViewNames.POS) ){
+        if (!record.hasView(ViewNames.POS)) {
             throw new AnnotatorException("Missing required view POS");
         }
-        if (!record.hasView(ViewNames.DEPENDENCY_STANFORD)){
+        if (!record.hasView(ViewNames.DEPENDENCY_STANFORD)) {
             throw new AnnotatorException("Missing required view DEPENDENCY_STANFORD");
         }
-        if (!record.hasView(ViewNames.SHALLOW_PARSE)){
+        if (!record.hasView(ViewNames.SHALLOW_PARSE)) {
             throw new AnnotatorException("Missing required view SHALLOW_PARSE");
         }
-        mentionAnnotator.addView(record);
+        if (!record.hasView(ViewNames.MENTION)) {
+            // TODO: show error messages if the mentions are not typed.
```

Comment:
What you can do here (outside this 'if' of course) is something like

```Java
else{
    if (record.getView(ViewNames.MENTION).getConstituents().get(0).getAttribute("EntityType").equals("MENTION")){
      //The model does not have type
    }
}

Reply:
Good point.

Reply:
Done.